### DISCUSSION TITLE
How do I install OpenModeliva v1.22 for Linux?

### DISCUSSION THREAD

**Discussion by ejperezHI** [OpenModelica/OpenModelica#11548] [Q&A] [answered]
Hello everyone,

How can I install the new 1.22 version of OpenModelica in my Ubutnu 20.04.6 LTS computer?

- This doesn't work (although it doesn't give me any errors): https://openmodelica.org/download/download-linux/
- Here I cannot find the package to download the stable version for my Ubuntu 20.04.6 LTS: https://build.openmodelica.org/

Thank you a lot in advance,

**Comment by AnHeuermann** [comment]
What commands did you run exactly (post them here) and do you have an older version of OpenModelica installed?

Running the install commands on a clean Ubuntu Focal works for me:

1. Get the OpenModelica public key that was used to sign OpenModelica and the tools to do so. So you are telling your OS to trust pacakges coming from OpenModelica.

    ```bash
    sudo apt-get update
    sudo apt-get install ca-certificates curl gnupg
    sudo curl -fsSL http://build.openmodelica.org/apt/openmodelica.asc | \
      sudo gpg --dearmor -o /usr/share/keyrings/openmodelica-keyring.gpg
    ```

2. Tell apt where to look for OpenModelica and what key was used to sign the packages:

    ```bash
    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/openmodelica-keyring.gpg] \
      https://build.openmodelica.org/apt \
      $(cat /etc/os-release | grep "\(UBUNTU\\|DEBIAN\\|VERSION\)_CODENAME" | sort | cut -d= -f 2 | head -1) \
      release" | sudo tee /etc/apt/sources.list.d/openmodelica.list
    ```

3. Update the package index, so the location added in step 2 can now be found as well. Install OpenModelica and all dependencies.
  
    ```bash
    sudo apt update
    sudo apt install openmodelica
    ```

4. Test omc:

    ```bash
    $ omc --version
    OpenModelica 1.22.0 
    ```

------------------

Check the content of `/etc/apt/sources.list.d/openmodelica.list`. If you have multiple entries for OpenModelica you can remove the older ones or remove `/etc/apt/sources.list.d/openmodelica.list` completely and re-run the second and third step from above. It will be re-created the file.

My `/etc/apt/sources.list.d/openmodelica.list`:
```
deb [arch=amd64 signed-by=/usr/share/keyrings/openmodelica-keyring.gpg]   https://build.openmodelica.org/apt   focal   release
```

--------

If all of this fails what is the error message you get?

------------------

You could try to remove OpenModelica and install it again step 3 or steps 2 and 3:

```
sudo apt update
sudo apt remove openmodelica
```

----------------------

> [!WARNING]
> Removing random packages and all dependencies can destroy all sort of things. Don't do it if you don't know what you are doing...

You can try to purge OpenModelica and all its dependencies and start again.

```bash
sudo aptitude purge openmodelica
sudo apt autoremove
```

**Reply by ejperezHI** [reply]
Thank you a lot!
Very helpful. I think that I finally managed to install OpenModelica 1.22. 

However it may not be well installed..., when I open OpenModelica and navegate through the left tree, I don't see the logos of each type of element (electrical, fluids...etc), appearing a "P" instead, and on the right hand side I keep getting "scripting errors". This isn't supposed to happen, right? At least this didn't happen to me in my previous version of OpenModelica.
![image](https://github.com/OpenModelica/OpenModelica/assets/131266350/bcd237be-45ca-4a6d-a47d-99ed710ddaba)
How can I solve this situation?
Thank you in advance!

**Reply by AnHeuermann** [reply]
No idea. Maybe remove the Modelica libraries installed with the package manager and re-install them. They are in `/home/USERNAME/.openmodelica/libraries/` and can be removed and than re-install them with the package manager.

**Reply by perost** [reply]
You might be missing the `omc-common` package which contains files that the compiler needs in order to be able to deal with e.g. graphical annotations.

**Reply by ejperezHI** [reply]
Thank you all for your replies,

I just deleted (I think) all the openmodelica stuff from my computer, including that directory @AnHeuermann .

It sounds fair @perost . But...how do I install it and more precisely, how do I know what to install exactly in order to not left anything behind?

Thanks,

**Reply by perost** [reply, marked as answer]
Check if you have the `omc-common` package installed, otherwise install it. If you already do have it you might need to reinstall it.

**Reply by ejperezHI** [reply]
Thanks,
Where should be that be installed? Within `../.openmodelica/libraries` ? And where do I can find omc-common to download it?

**Reply by perost** [reply]
It's just a normal package, so `sudo apt install omc-common`.

**Reply by ejperezHI** [reply]
Thanks for your reply,
That was the thing! I didn't have the `omc-common` package updated. Once updated, I can read all the symbols perfectly

**Comment by casella** [comment]
@perost, @sjoelund, I just checked the [Linux installation instructions](https://openmodelica.org/download/download-linux/), but it doesn't explicitly mention the `omc-common` package. Is this expected to be automatically installed when doing `sudo apt install openmodelica`, or should we add `sudo apt install omc-common` to the list of commands to be issued?

**Reply by perost** [reply]
`omc-common` is a dependency of `openmodelica`, so it should normally be installed automatically.

**Reply by casella** [reply]
This is what scares me a bit about all Linux stuff, like compiling from sources or installing from apt. _Normally_ everything works well, but if it doesn't, then all of a sudden you need to be a guru to fix it... 😅

**Reply by adrpo** [reply]
For me it worked with:

`sudo apt-get --reinstall install omc-common`
